packages/insured-bridge-relayer/src/index.ts

```diff
@@ -113,7 +113,14 @@ export async function run(logger: winston.Logger, l1Web3: Web3): Promise<void> {
 
           if (config.botModes.l2FinalizerEnabled) await crossDomainFinalizer.checkForBridgeableL2TokensAndBridge();
           else logger.debug({ at: "AcrossRelayer#CrossDomainFinalizer", message: "Cross domain finalizer disabled" });
+
+          // Each of the above code blocks could have produced transactions. If they did, their promises are stored
+          // in the executed transactions array. The method below awaits all these transactions to ensure they are
+          // correctly included in a block. if any submitted transactions contains an error then a log is produced.
+          await processTransactionPromiseBatch(relayer.getExecutedTransactions(), logger);
```

Comment:
note here: at the end of the execution loop we process the array of transactions by awaiting each. if any failed it will produce a log accordingly.